src/media/js/addon/reducers/addon.js

```diff
@@ -73,6 +73,17 @@ export default function addonReducer(state=initialState, action) {
       return newState;
     }
 
+    case dashboardActions.DELETE_OK: {
+      /*
+        Mark add-on as deleted.
+
+        payload (string) -- addonSlug
+      */
+      const newState = _.cloneDeep(state);
+      newState.addons[action.payload].deleted = true;
```

Comment:
what's nice is that if the user navs back to this page somehow, they will get a nice message that the add-on was deleted rather than a 404